internal/runtime/runtime.go

```diff
@@ -37,8 +38,31 @@ func (gr GolangRuntime) ProcessEvent(edgexcontext *appcontext.Context, envelope
 	edgexcontext.LoggingClient.Debug("Processing Event: " + strconv.Itoa(len(gr.Transforms)) + " Transforms")
 	var event models.Event
 
-	if err := json.Unmarshal([]byte(envelope.Payload), &event); err != nil {
-		edgexcontext.LoggingClient.Error("Expected JSON EdgeX Event: " + err.Error())
+	switch envelope.ContentType {
+	case clients.ContentTypeJSON:
+		if err := json.Unmarshal([]byte(envelope.Payload), &event); err != nil {
+			edgexcontext.LoggingClient.Error("Unable to JSON unmarshal  EdgeX Event: "+err.Error(), clients.CorrelationHeader, envelope.CorrelationID)
+			return nil
+		}
+
+		// Needed for Marking event as handled
+		edgexcontext.EventId = event.ID
+
+	case clients.ContentTypeCBOR:
+		event := models.Event{}
+
+		x := codec.CborHandle{}
+		err := codec.NewDecoderBytes([]byte(envelope.Payload), &x).Decode(&event)
+		if err != nil {
+			edgexcontext.LoggingClient.Error("Unable to CBOR unmarshal EdgeX Event: "+err.Error(), clients.CorrelationHeader, envelope.CorrelationID)
+			return nil
+		}
+
+		// Needed for Marking event as handled
+		edgexcontext.EventChecksum = envelope.Checksum
```

Comment:
I don't see anywhere that the EventClient `MarkPushed` methods are being called. So in either case, it doesn't look like this code marks events as pushed yet.

Reply:
Correct, @rsdmike is working on that in a separate PR. This setups the context with the information he will need to call the 'MarkPushed' methods